compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -557,6 +590,43 @@ tyConAnchor DocCtx{..} tycon = do
             | otherwise = typeAnchor
     Just (anchorFn mod name)
 
+-- | Extract a potentially qualified typename from a TyCon.
+tyConTypename :: DocCtx -> TyCon -> Typename
+tyConTypename DocCtx{..} tycon =
+    let ExtractOptions{..} = dc_extractOptions
+        ghcName = tyConName tycon
+        qualify =
+            case eo_qualifyTypes of
+                QualifyTypesAlways -> True
+                QualifyTypesInPackage -> nameIsHomePackageImport dc_ghcMod ghcName
+                QualifyTypesNever -> False
+
+        moduleM = guard qualify >> nameModule_maybe ghcName
+        modNameM = getModulename <$> moduleM
+        simplifyModName
+            | eo_simplifyQualifiedTypes = dropCommonModulePrefix dc_modname
+            | otherwise = id
+        prefix = maybe "" ((<> ".") . unModulename . simplifyModName) modNameM
+    in Typename (prefix <> packName ghcName)
+
+-- | Drop common module name prefix, returning the second module name
+-- sans the module prefix it has in common with the first module name.
+-- This will not return an empty module name however (unless given an
+-- empty module name to start).
+--
+-- This function respects the atomicity of the module names between
+-- periods. For instance @dropCommonModulePrefix "Foo.BarBaz" "Foo.BarSpam"@
+-- will evaluate to @"BarSpam"@, not @"Spam"@.
+dropCommonModulePrefix :: Modulename -> Modulename -> Modulename
```

Comment:
Is `dc_modname` the same as the root module of your DALF? In that case, I am not sure this option is very useful as a lot of people seem to just call that module `LibraryModules`. Maybe an option that strips away everything but the last component would be more useful? (I realize that this makes things potentially ambiguous.)

Reply:
`dc_modname` is the name of the current module.

> Maybe an option that strips away everything but the last component would be more useful?

That's definitely another possibility.

Reply:
Oh I see, in that case the current behavior seems fine.